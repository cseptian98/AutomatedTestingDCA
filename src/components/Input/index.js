import React, {useState} from 'react';
import {View, TextInput, TouchableOpacity, StyleSheet} from 'react-native';
import axiosConfig from '../../api/BaseConfig';

const InputTodo = () => {
  const [title, setTitle] = useState('');

  const doCreate = () => {
    const onSuccess = ({data}) => {
      console.log('debug success', data);
    };

    const onFailure = error => {
      console.log('debug error', error.response.data);
    };

    axiosConfig.post('api/TodoLists', {title}).then(onSuccess).catch(onFailure);
  }

  return (
    <View style={styles.container}>
      <View style={styles.inputView}>
        <TextInput
          style={styles.textInput}
          placeholder="Add Todo List..."
          placeholderTextColor="#000"
          onChangeText={title => setTitle(title)}
        />
      </View>
      <TouchableOpacity style={styles.button} onPress={doCreate}>
        {/* ICON */}
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    marginTop: 20,
  },
  inputView: {
    borderRadius: 10,
    flexDirection: 'row',
  },
  textInput: {
    width: 300,
    padding: 10,
    marginLeft: 10,
    marginRight: 10,
    marginBottom: 20,
    borderRadius: 10,
    backgroundColor: '#FFD500',
  },
  button: {
    width: 50,
    marginBottom: 20,
    borderRadius: 50,
    backgroundColor: '#FFD500',
  },
});

export default InputTodo;
