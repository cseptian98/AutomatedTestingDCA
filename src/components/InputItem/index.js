import React, {useState} from 'react';
import {View, TextInput, TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import axiosConfig from 'api/BaseConfig';
import styles from './InputItems.styles';

const InputItem = () => {
  const listId = 1;
  const [title, setTitle] = useState('');

  const createNewItem = () => {
    const onSuccess = ({data}) => {
      console.log('debug success', data);
      setTitle('');
    };

    const onFailure = error => {
      console.log('debug error', error);
    };

    axiosConfig
      .post('api/TodoItems', {listId, title})
      .then(onSuccess)
      .catch(onFailure);
  };

  return (
    <View style={styles.container}>
      <View style={styles.inputView}>
        <TextInput
          style={styles.textInput}
          placeholder="Add Todo Item..."
          placeholderTextColor="#000"
          onChangeText={title => setTitle(title)}
        />
      </View>
      <TouchableOpacity style={styles.button} onPress={createNewItem}>
        <Icon name="add" size={24} />
      </TouchableOpacity>
    </View>
  );
};

export default InputItem;
