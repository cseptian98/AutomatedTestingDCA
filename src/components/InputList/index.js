import React, {useState} from 'react'
import {View, TextInput, TouchableOpacity, Image} from 'react-native'
import Icon from 'react-native-vector-icons/MaterialIcons'
import styles from './InputList.styles'
import axiosConfig from 'api/BaseConfig'

const InputList = ({createList}) => {
  const [title, setTitle] = useState('')

  // const createNewList = () => {
  //   const onSuccess = ({data}) => {
  //     console.log('debug success', data)
  //     setTitle('')
  //   }

  //   const onFailure = error => {
  //     console.log('debug error', error)
  //   }

  //   axiosConfig.post('api/TodoLists', {title}).then(onSuccess).catch(onFailure)
  // }

  return (
    <View style={styles.container}>
      <View>
        <Image
          style={styles.image}
          source={require('assets/images/list.png')}
        />
      </View>
      <View style={styles.inputView}>
        <TextInput
          style={styles.textInput}
          placeholder="Add Todo List..."
          placeholderTextColor="#FFF"
          onChangeText={title => setTitle(title)}
        />
        <TouchableOpacity
          style={styles.button}
          onPress={() => createList(title)}>
          <Icon name="add" size={30} color="#FFF" />
        </TouchableOpacity>
      </View>
    </View>
  )
}

export default InputList
