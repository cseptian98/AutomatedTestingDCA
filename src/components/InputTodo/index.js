import React from 'react';
import {View, TextInput, TouchableOpacity, StyleSheet} from 'react-native';

const InputTodo = () => {
  return (
    <View style={styles.container}>
      <View style={styles.inputView}>
        <TextInput
          style={styles.textInput}
          placeholder="Add Todo List..."
          placeholderTextColor="#000"
        />
      </View>
      <TouchableOpacity style={styles.button}>
        {/* ICON */}
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  inputView: {
    borderRadius: 10,
    flexDirection: 'row',
  },
  textInput: {
    width: 300,
    padding: 10,
    marginLeft: 10,
    marginRight: 10,
    marginBottom: 20,
    borderRadius: 10,
    backgroundColor: '#FFD500',
  },
  button: {
    width: 50,
    marginBottom: 20,
    borderRadius: 50,
    backgroundColor: '#FFD500',
  },
});

export default InputTodo;
