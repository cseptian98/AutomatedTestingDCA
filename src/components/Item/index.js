import React from 'react'
import {TouchableOpacity, Text} from 'react-native'
import styles from './Item.styles'

const Item = ({title, onDelete}) => (
  <TouchableOpacity style={styles.item} onPress={onDelete}>
    <Text style={styles.title}>{title}</Text>
  </TouchableOpacity>
)

export default Item
