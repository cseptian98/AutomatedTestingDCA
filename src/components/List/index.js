import React from 'react';
import {TouchableOpacity, Text, View} from 'react-native';
import styles from './List.styles';

const List = ({title, onDelete}) => (
  <TouchableOpacity style={styles.item} onPress={onDelete}>
    <Text style={styles.title}>{title}</Text>
  </TouchableOpacity>
);

export default List;
