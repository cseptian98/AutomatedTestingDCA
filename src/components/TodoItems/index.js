import React, {useState, useEffect} from 'react';
import {SafeAreaView, FlatList, StyleSheet} from 'react-native';
import List from 'components/List';
import axiosConfig from 'api/BaseConfig';

const TodoItems = () => {
  const ListId = 1;
  const [item, setItem] = useState([]);

  useEffect(() => {
    getTodoItems();
  }, []);

  const getTodoItems = () => {
    const onSuccess = ({data}) => {
      console.log('debug success', data.items);
      setItem(data.items);
    };

    const onFailure = error => {
      console.log('debug error', error.response.data);
    };

    axiosConfig
      .get('api/TodoItems', {params: {ListId}})
      .then(onSuccess)
      .catch(onFailure);
  };

  const renderItem = ({item}) => <List title={item.title} />;

  return (
    <SafeAreaView style={styles.container}>
      <FlatList
        data={item}
        renderItem={renderItem}
        keyExtractor={item => item.id}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FFF',
  },
});

export default TodoItems;
