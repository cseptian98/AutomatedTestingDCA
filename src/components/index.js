import FormUpdateItem from './TodoItems/FormUpdateItem/FormUpdateItem'
import TodoLists from './TodoLists/TodoLists'
import TodoItems from './TodoItems/TodoItems'
import User from './User/User'

export {FormUpdateItem, TodoLists, TodoItems, User}