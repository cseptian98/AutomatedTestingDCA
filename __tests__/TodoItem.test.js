import React from 'react'
import {fireEvent, render, waitFor} from '@testing-library/react-native'
import {TodoItems} from 'components'
import {
  TEST_ID_IMAGE_TODOITEM,
  TEST_ID_TEXT_INPUT_ITEM,
  TEST_ID_BUTTON_SUBMIT_ITEM,
} from 'constants'

let todoItem
let componentTodoItems
let inputItem
let buttonSubmitItem
const params = {url: 'api/TodoItems'}
const dispatch = jest.fn()

describe('TodoItems screen', () => {
  beforeEach(() => {
    componentTodoItems = render(<TodoItems route={{params}} navigation={{dispatch}}/>)

    todoItem = 'Shoes'

    inputItem = componentTodoItems.getByTestId(TEST_ID_TEXT_INPUT_ITEM)
    buttonSubmitItem = componentTodoItems.getByTestId(
      TEST_ID_BUTTON_SUBMIT_ITEM,
    )
  })

  it('should show todoitems screen', () => {
    expect(componentTodoItems.getByTestId(TEST_ID_IMAGE_TODOITEM)).toBeTruthy()
  })

  it('should change text input value', async () => {
    await waitFor(() => {
      fireEvent.changeText(inputItem, todoItem)
    })
    expect(inputItem.props.value).toBe(todoItem)
  })
})
