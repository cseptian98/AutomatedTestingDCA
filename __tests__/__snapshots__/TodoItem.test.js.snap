// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItems screen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 10,
      }
    }
  >
    <View>
      <Image
        accessibilityLabel="TEST_ID_IMAGE_TODOITEM"
        source={
          Object {
            "testUri": "../../../src/assets/images/item.png",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "height": 120,
            "justifyContent": "center",
            "width": 160,
          }
        }
        testID="TEST_ID_IMAGE_TODOITEM"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 10,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <TextInput
        accessibilityLabel="TEST_ID_TEXT_INPUT_ITEM"
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Add Todo Item..."
        placeholderTextColor="#FFF"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#0D47A1",
            "borderRadius": 10,
            "color": "#FFF",
            "marginBottom": 10,
            "marginLeft": 16,
            "padding": 10,
            "width": 200,
          }
        }
        testID="TEST_ID_TEXT_INPUT_ITEM"
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessibilityLabel="TEST_ID_BUTTON_SUBMIT_ITEM"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#0D47A1",
            "borderRadius": 10,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 16,
            "opacity": 1,
            "width": 200,
          }
        }
        testID="TEST_ID_BUTTON_SUBMIT_ITEM"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#FFF",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          î……
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    accessible={false}
  >
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
